src/routes/InstancesPage/InstancesPage.js

```diff
@@ -373,9 +378,7 @@ function InstancesPage() {
       <Main>
         <Card>{content}</Card>
         <CreateInstanceModal
-          isOpen={
-            !!creatingInstance || (instances.length === 0 && !isTableLoading)
-          }
+          isOpen={!!creatingInstance || hasNoInstances}
```

Comment:
Without the check for `Object.keys(filters).length === 0`, as soon as a filter is applied the modal opened and was not closable.